compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2000,33 +2000,28 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         //
         // For the "output" lifetime parameters, we just want to
         // generate `'_`.
-        let mut generic_args: Vec<_> = lifetime_params[..input_lifetimes_count]
-            .iter()
-            .map(|&(span, hir_name)| {
-                // Input lifetime like `'a` or `'1`:
-                GenericArg::Lifetime(hir::Lifetime {
-                    hir_id: self.next_id(),
-                    span,
-                    name: hir::LifetimeName::Param(hir_name),
-                })
-            })
-            .collect();
-        generic_args.extend(lifetime_params[input_lifetimes_count..].iter().map(|&(span, _)|
-            // Output lifetime like `'_`.
+        let arena = self.arena;
+        let generic_args = lifetime_params.iter().enumerate().map(|(i, &(span, hir_name))| {
+            // Input lifetime like `'a` or `'1`:
             GenericArg::Lifetime(hir::Lifetime {
                 hir_id: self.next_id(),
                 span,
-                name: hir::LifetimeName::Implicit,
-            })));
-        let generic_args = self.arena.alloc_from_iter(generic_args);
+                name: if i < input_lifetimes_count {
```

Comment:
It might make sense to use `chain` here instead of branching on every lt param, though I expect this to not really matter

Reply:
I think I had ownership issues with chain, but I'll check

Reply:
I hate this piece of code. This piece of code allocates a vector, then copies that vector in the allocator into another vector, then into the arena... Oh well, if only I could reuse `alloc_slice`, but no, `GenericArgs` is not `Copy`. 😅